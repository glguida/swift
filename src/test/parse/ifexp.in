random PrepLevel Prep(City c) {
  if (First == c) then ~ Categorical({High -> 0.5, Low -> 0.5})
  else ~ Categorical({High -> 0.8, Low -> 0.2})
};