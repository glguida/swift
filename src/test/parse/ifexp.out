(BlogProgram:
  (args:
    (FuncDecl:
      :kind random
      :type PrepLevel
      :func Prep
      (args:
        (VarDecl: (: City c ) )
      )
      expr:
        (IfExpr:
          :cond
            (OpExpr:
              :op 6
              :left
                (VarRef: First )
              :right
                (VarRef: c )
            )
          :then
            (DistrExpr:
              :distr Categorical
              (args:
                (MapExpr:
                  :from#0
                    (VarRef: High )
                  :to#0
                    (DoubleLiteral: 0.500000 )
                  :from#1
                    (VarRef: Low )
                  :to#1
                    (DoubleLiteral: 0.500000 )
                )
              )
            )
          :else
            (DistrExpr:
              :distr Categorical
              (args:
                (MapExpr:
                  :from#0
                    (VarRef: High )
                  :to#0
                    (DoubleLiteral: 0.800000 )
                  :from#1
                    (VarRef: Low )
                  :to#1
                    (DoubleLiteral: 0.200000 )
                )
              )
            )
        )
    )
  )
)
